libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/VoiceInstructionLoader.java

```diff
@@ -0,0 +1,198 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import com.mapbox.api.directions.v5.models.LegStep;
+import com.mapbox.api.directions.v5.models.VoiceInstructions;
+import com.mapbox.api.speech.v1.MapboxSpeech;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.Cache;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+@AutoValue
+public abstract class VoiceInstructionLoader {
+  private static final int NUMBER_TO_CACHE = 3;
+  private static VoiceInstructionLoader instance = null;
+
+  /**
+   * Returns the singleton instance of VoiceInstructionLoader. It must first be initialized through
+   * a builder.
+   *
+   * @return current instance if it's initialized, or null
+   */
+  public static synchronized VoiceInstructionLoader getInstance() {
```

Comment:
https://github.com/mapbox/mapbox-navigation-android/pull/751#discussion_r179159458
Is this strictly necessary? Can't the different classes that use it create an instance of it when need it?

Reply:
@Guardiola31337 they won't share the same cache if they aren't using the singleton